### PR TITLE
[MyCollection] Add artist search

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: **Enhancement / Feature**

### Description

This PR adds a new artist search autosuggest input to MyCollection by way of modifying search AutoSuggest in three ways: 

1) Updates it to accept different [kinds of `entities`](https://github.com/damassi/metaphysics/blob/6b047aaf370d1de81caa835800a8300f5b4c5c2f/src/schema/v2/search/SearchEntity.ts)
1) Adds a new `showResultType` prop for disabling the entity type label returned from MP
1) Adds a new `onResultPress` prop for handling presses to the result. 

This seems to be working good, but looking for feedback from MX team on approach. 

There's still a bit of layout faffing we need to do around this but can take care of that in a follow-up; this just gets the functionality in. 

### Test Plan

Will update Search tests after feedback from @ds300 / @ashfurrow. 

### Screenshots

sug

#skip_new_tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:

- `src/lib/Scenes/Consignments/v2/Screens/MyCollectionAddArtwork/Components/__tests__/ArtistAutosuggest-tests.tsx`
- `src/lib/Scenes/Consignments/v2/Screens/MyCollectionAddArtwork/Components/__tests__/ArtistSearchResult-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Scenes/Search/AutosuggestResults.tsx [hunk 1]
```diff
@@ -125,14 +135,14 @@ const AutosuggestResultsContainer = createPaginationContainer(
...
-          mode: AUTOSUGGEST
-          first: $count
-          after: $cursor
-          entities: [ARTIST, ARTWORK, FAIR, GENE, SALE, PROFILE, COLLECTION]
-        ) @connection(key: "AutosuggestResults_results") {
+        @argumentDefinitions(
+          query: { type: "String!" }
+          count: { type: "Int!" }
+          cursor: { type: "String" }
+          entities: { type: "[SearchEntity]", defaultValue: [ARTIST, ARTWORK, FAIR, GENE, SALE, PROFILE, COLLECTION] }
```
Comment (damassi) [inline src/lib/Scenes/Search/AutosuggestResults.tsx:142]: Main change is moving these into a default value

File: src/lib/Scenes/Consignments/v2/Screens/MyCollectionAddArtwork/Components/ArtistAutosuggest.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      {!!artistSearchResult ? (
+        <>
+          <Sans size="3t" mt="2px">
+            Artist
+          </Sans>
+          <Spacer mt={0.5} />
+          <ArtistSearchResult result={artistSearchResult} />
+        </>
+      ) : (
+        <ProvideRecentSearches>
```
Comment (damassi) [inline src/lib/Scenes/Consignments/v2/Screens/MyCollectionAddArtwork/Components/ArtistAutosuggest.tsx:29]: This is needed for the autosuggest, even though we don't use its functionality.

File: src/lib/Scenes/Search/Search.tsx [hunk 3]
```diff
@@ -1,33 +1,30 @@
...
 
 const SearchPage: React.FC = () => {
-  const input = useRef<Input>(null)
   const [query, setQuery] = useState("")
-  const queryRef = useRef(query)
-  queryRef.current = query
   const { recentSearches } = useRecentSearches()
   const { trackEvent } = useTracking()
-
+  const searchProviderValues = useSetupSearchContext(query)
```
Comment (pepopowitz) [inline src/lib/Scenes/Search/Search.tsx:19]: **suggestion [non-blocking]**: We're consistently storing the result of `useSetupSearchContext` in a variable named `searchProviderValues`. Does this suggest that a better name for `useSetupSearchContext` would be `useSearchProviderValues`?

File: src/lib/Scenes/Search/SearchResult.tsx [hunk 4]
```diff
@@ -11,37 +11,53 @@ import { AutosuggestResult } from "./AutosuggestResults"
...
-            ? Schema.ActionNames.ARAnalyticsSearchRecentItemSelected
-            : Schema.ActionNames.ARAnalyticsSearchItemSelected,
-          query: query.current,
-          selected_object_type: result.displayType,
-          selected_object_slug: result.slug,
-        })
+        if (onResultPress) {
+          onResultPress(result)
+        } else {
+          inputRef.current?.blur()
```
Comment (pepopowitz) [inline src/lib/Scenes/Search/SearchResult.tsx:44]: **question [non-blocking]**: Is the `blur` effect something we will want to happen even when there is an `onResultPress` provided?

File: src/lib/Scenes/Consignments/v2/State/artworkModel.tsx [hunk 5]
```diff
@@ -47,6 +51,17 @@ export const artworkModel: ArtworkModel = {
     state.formValues = input
   }),
 
+  setArtistSearchResult: action((state, artistSearchResult) => {
+    state.formValues = {
+      ...state.formValues,
+      artistSearchResult,
+    }
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/lib/Scenes/Consignments/v2/State/artworkModel.tsx:58]: Oh you know, I think so! Still not totally used to this immer way of doing things. Will update.

File: src/lib/Scenes/Search/Search.tsx (hunk 3 above)
Comment (damassi) [reply src/lib/Scenes/Search/Search.tsx:19]: Ohh yup 👍 that feels much less ackward.

File: src/lib/Scenes/Search/SearchResult.tsx (hunk 4 above)
Comment (damassi) [reply src/lib/Scenes/Search/SearchResult.tsx:44]: My only addition here was the addition of the callback on line 42 (and the logical branch)

**Pull request merged**